editor/src/components/canvas/canvas-strategies/escape-hatch-strategy.tsx

```diff
@@ -416,3 +405,105 @@ function collectHighlightCommand(
   }, canvasState.selectedElements)
   return showOutlineHighlight('transient', [...siblingFrames, ...draggedFrames])
 }
+
+function findAbsoluteDescendants(
+  selectedElements: Array<ElementPath>,
+  metadata: ElementInstanceMetadataMap,
+): Array<ElementPath> {
+  const isSelectionInsideFocusedComponent = selectedElements.every((path) =>
+    EP.isInsideFocusedComponent(path),
+  )
+  return mapDropNulls((element) => {
+    const elementPath = element.elementPath
+    const isInsideFocusedComponent = EP.isInsideFocusedComponent(elementPath)
+    const allInsideComponentOrOutside =
+      isInsideFocusedComponent === isSelectionInsideFocusedComponent // if the selection and the children are both inside of the focused component
+
+    const sortedAncestors = EP.getOrderedPathsByDepth(selectedElements).filter((selection) =>
+      EP.isDescendantOf(elementPath, selection),
+    )
+    if (
+      MetadataUtils.isPositionAbsolute(element) &&
+      !EP.isRootElementOfInstance(elementPath) &&
+      sortedAncestors.length > 0 &&
+      allInsideComponentOrOutside
+    ) {
+      const closestAncestor = sortedAncestors[0]
+      const containingBlockPath = MetadataUtils.findContainingBlock(metadata, elementPath)
+      const closestAncestorContainingBlockPath = MetadataUtils.findContainingBlock(
+        metadata,
+        closestAncestor,
+      )
+      if (EP.pathsEqual(containingBlockPath, closestAncestorContainingBlockPath)) {
```

Comment:
I honestly have no idea what the reason is for this check. Please can you add some commenting here.